node/src/actors/chain_manager/mining.rs

```diff
@@ -285,24 +284,27 @@ impl ChainManager {
                 .flatten()
                 .and_then(move |(vrf_proof, reveal_value)| {
                     let vrf_proof_dr = DataRequestEligibilityClaim { proof: vrf_proof };
-                    // Create commitment transaction
-                    let commit_body =
-                        create_commit_body(dr_pointer, reveal_value.clone(), vrf_proof_dr);
-                    sign_transaction(&commit_body, 1)
-                        .map_err(|e| log::error!("Couldn't sign commit body: {}", e))
-                        .and_then(move |commit_signatures| {
-                            let reveal_body = create_reveal_body(dr_pointer, reveal_value, own_pkh);
-
-                            sign_transaction(&reveal_body, 1)
-                                .map(|reveal_signatures| {
+
+                    let reveal_body = RevealTransactionBody::new(dr_pointer, reveal_value, own_pkh);
+
+                    sign_transaction(&reveal_body, 1)
```

Comment:
What do you think about adding a `create_commit_and_reveal` function to `signature_mngr`?